skpro/regression/linear/_glm.py

```diff
@@ -0,0 +1,444 @@
+        changes as well.
+
+    iteration_ : int
+        The number of iterations that fit has run. Initialized at 0.
+        Only available after fit is called
+
+    family_ : family class instance
+        he distribution family of the model. Can be any family
+        in statsmodels.families. Default is Gaussian.
+
+    mu_ : ndarray
+        The mean response of the transformed variable. mu is the value of the
+        inverse of the link function at lin_pred, where lin_pred is the linear
+        predicted value of the WLS fit of the transformed variable. mu is only
+        available after fit is called. See statsmodels.families.family.fitted
+        of the distribution family for more information.
+
+    n_trials_ : ndarray
+        Note that n_trials is a reference to the data so that if data is
+        already an array and it is changed, then n_trials changes as well.
+        n_trials is the number of binomial trials and only available with that
+        distribution. See statsmodels.families.Binomial for more information.
+
+    scale_ : float
+        The estimate of the scale / dispersion of the model fit.
+        Only available after fit is called. See GLM.fit and GLM.estimate_scale
+        for more information.
+
+    scaletype_ : str
+        The scaling used for fitting the model. This is only available
+        after fit is called. The default is None. See GLM.fit for
+        more information.
+
+    weights_ : ndarray
+        The value of the weights after the last iteration of fit.
+        Only available after fit is called. See statsmodels.families.family
+        for the specific distribution weighting functions.
+
+    glm_fit_ : GLM
+        fitted generalized linear model
+
+    fit_history_ : dict
+        Contains information about the iterations.
+        Its keys are iterations, deviance and params. Only available after
+        fit is called.
+
+    model_ : class instance
+        Pointer to GLM model instance that called fit.
+
+    nobs_ : float
+        The number of observations n. Only available after fit is called.
+
+    normalized_cov_params_ : ndarray
+        For Gaussian link: This is the p x p normalized covariance of the
+        design / exogenous data. This is approximately equal to (X.T X)^(-1)
+
+    params_ : ndarray
+        The coefficients of the fitted model. Note that interpretation of the
+        coefficients often depends on the distribution family and the data.
+
+    pvalues_ : ndarray
+        The two-tailed p-values for the parameters.
+
+    scale_ : float
+        The estimate of the scale / dispersion for the model fit.
+        See GLM.fit and GLM.estimate_scale for more information.
+
+    stand_errors_ : ndarray
+        The standard errors of the fitted GLM.
+    """
+
+    _tags = {
+        "authors": ["julian-fong"],
+        "maintainers": ["julian-fong"],
+        "python_version": None,
+        "python_dependencies": None,
```

Comment:
this tag should have `statsmodels`, so `skpro` knows this requires the dependency.

Reply:
I added the statsmodel dependency and renamed the module to GLMRegressor. I can refactor some of the code to include all of the available links inside the statsmodels module, but I think another member is currently working on the glm with count link

Reply:
I think that's a different estimator, it's `sklearn`'s Poisson regressor, that's different from `statsmodels` `GLM` with Poisson link: https://github.com/sktime/skpro/pull/213/files

Further, I'd expect more link functions to be added to this class.